src/actions/coordinator.ts

```diff
@@ -1,198 +1,116 @@
-      //   await this.getTransactionReceipt(closeTx)
+      /**
+       * We divide all uint values by 10**18, in order to convert them to JS numbers. This induces some loss of precision,
+       * but since we are currently using DAI as the ERC20 token, this isn't a big problem. We are also dividing the
+       * ratios by 10**20, then converting them to numbers, and then dividing again by 10**7. This ultimately divides
+       * the ratios by 10**27, which is the precision of these values on contract. However, BN.js doesn't support decimals,
+       * so we basically limit the ratios to 7 decimals here.
+       */
+      const valueBase = new BN(10).pow(new BN(18))
+      const ratioBase = new BN(10).pow(new BN(20))
 
-      //   if (!(await this.contract('COORDINATOR').submissionPeriod)) return
-      // }
+      const reserveBN = (await coordinator.epochReserve()).toBN()
+      const reserve = reserveBN.isZero() ? 0.0 : reserveBN.div(valueBase).toNumber()
 
-      // const state = {
-      //   reserve: await coordinator.epochReserve, // coordinator.epochReserve
-      //   netAssetValue: await coordinator.epochNAV, // coordinator.epochNAV
-      //   seniorAsset: await coordinator.epochSeniorAsset, // coordinator.epochSeniorDebt (to be added)
-      //   minTinRatio: await tinlake.getMinJuniorRatio(), // 1 - maxSeniorRatio on the assessor
-      //   maxTinRatio: 0, // 1 - mSeniorRatio on the assessor
-      //   maxReserve: 0, // assessor.maxReserve
-      // }
+      const netAssetValueBN = (await coordinator.epochNAV()).toBN()
+      const netAssetValue = netAssetValueBN.isZero() ? 0.0 : netAssetValueBN.div(valueBase).toNumber()
 
-      // const orderState = coordinator.order
+      const seniorAssetBN = (await coordinator.epochSeniorAsset()).toBN()
+      const seniorAsset = seniorAssetBN.isZero() ? 0.0 : seniorAssetBN.div(valueBase).toNumber()
 
-      // const solution = calculateOptimalSolution(state, orderState)
+      const maxDROPRatioBN = (await assessor.maxSeniorRatio()).toBN()
+      const minTinRatio = maxDROPRatioBN.isZero() ? 1.0 : 1.0 - maxDROPRatioBN.div(ratioBase).toNumber() / 10 ** 7
 
-      // Call submitSolution(solution)
+      const minDROPRatioBN = (await assessor.minSeniorRatio()).toBN()
+      const maxTinRatio = minDROPRatioBN.isZero() ? 1.0 : 1.0 - minDROPRatioBN.div(ratioBase).toNumber() / 10 ** 7
 
-      return Promise.resolve({
-        tinRedeem: 1,
-        dropRedeem: 2,
-        tinInvest: 3,
-        dropInvest: 4
-      })
+      const maxReserveBN = (await assessor.maxReserve()).toBN()
+      const maxReserve = maxReserveBN.isZero() ? 0.0 : maxReserveBN.div(valueBase).toNumber()
 
+      return { reserve, netAssetValue, seniorAsset, minTinRatio, maxTinRatio, maxReserve }
     }
 
-    // executeEpoch = () => void
+    solveEpoch = async () => {
+      const coordinator = this.contract('COORDINATOR')
+
+      if (!(await coordinator.submissionPeriod())) {
```

Comment:
just a small thing but maybe it's nice to make it explicit for readability here: `if (await.....) === false`

Reply:
Good idea!